### PR TITLE
Add CI workflow for running doctests

### PR DESCRIPTION
**Pull request by elliotwutingfeng**
Addresses #9 

- Added CI workflow for testing on windows, macos and ubuntu

I found that all tests were passing except on Ubuntu due to mismatched outputs, you can check the GitHub Actions [logs](https://github.com/elliotwutingfeng/dhash/actions/runs/3275505315) for more details.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  test:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+        python-version: ['2.7', '3.7', '3.8', '3.9', '3.10']
```
Comment (benhoyt) [inline .github/workflows/ci.yml]: Let's drop 3.7 at this point, just to keep the matrix slightly smaller.

File: .gitignore [hunk 2]
```diff
@@ -1,5 +1,162 @@
-__pycache__
-*.pyc
+# Byte-compiled / optimized / DLL files
```
Comment (benhoyt) [inline .gitignore]: I don't love these "everything" `.gitignore` files, let's just limit it to what we actually need for this project: perhaps all we need is `*.pyc`, `__pycache__/` for now?

File: dhash.py [hunk 3]
```diff
@@ -256,7 +256,9 @@ def load_image(filename):
     if len(args.filename) == 0:
         # NOTE: doctests require "wand" to be installed
         import doctest
-        doctest.testmod()
+        test_results = doctest.testmod()
+        if test_results.failed:
```
Comment (benhoyt) [inline dhash.py]: The docs seem to indicate it returns a tuple of `failure_count, test_count`. Maybe it's a namedtuple in the implementation, but it doesn't seem like we should rely on this. I think we should change to:

```python
failure_count, _ = doctest.testmod()
if failure_count:
    sys.exit(1)
```

**Review by benhoyt — COMMENTED** [review]
Thanks for this! Just a couple of minor comments.

Do you want to address the `ANTIALIAS` warning issue from #9 as well?

**Comment by benhoyt** [comment]
Do you have any suggestions for how to fix the test failures on Linux? It's odd that the numbers are off by one in some cases, given that it's the same architecture and Pillow/Wand versions. My guess is floating point rounding issues?

File: .gitignore (hunk 2 above)
Comment (elliotwutingfeng) [reply .gitignore]: Reverted!

File: dhash.py (hunk 3 above)
Comment (elliotwutingfeng) [reply dhash.py]: It is a named tuple, though this approach works too.

**Comment by elliotwutingfeng** [comment]
> Thanks for this! Just a couple of minor comments.
> 
> Do you want to address the `ANTIALIAS` warning issue from #9 as well?

I've replaced it with `PIL.Image.Resampling.LANCZOS`. Test results are the same as before.

**Comment by elliotwutingfeng** [comment]
> Do you have any suggestions for how to fix the test failures on Linux? It's odd that the numbers are off by one in some cases, given that it's the same architecture and Pillow/Wand versions. My guess is floating point rounding issues?

Strangely, the tests pass locally on my Ubuntu x64 and Manjaro x64 virtual machines, on both Python 2 and 3 (3.10.6).

**Comment by benhoyt** [comment]
Thanks, @elliotwutingfeng. I added a few more changes to allow older and new versions of Pillow, and "fixed" the tests.

**Comment by elliotwutingfeng** [comment]
I re-ran the old tests with this workflow (https://github.com/elliotwutingfeng/dhash/actions/runs/3280926289).

`ubuntu-latest` (`ubuntu-20.04` as of today) uses an older version of `libmagickwand-dev` than `ubuntu-22.04`, that may be causing the output differences. See below.

https://github.com/actions/runner-images/blob/main/images/linux/Ubuntu2004-Readme.md#installed-apt-packages
https://github.com/actions/runner-images/blob/main/images/linux/Ubuntu2204-Readme.md#installed-apt-packages

The old tests are successful on 22.04 just like windows and macos. Python 2 had to be manually installed using apt-get on ubuntu-22.04 as it is deprecated. I think the delta workaround is not necessary.

`ubuntu-latest` will be bumped to `ubuntu-22.04` by around [December 2022](https://github.com/actions/runner-images/issues/6399).

**Pull request merged**